internal/host/host.go

```diff
@@ -458,20 +468,29 @@ func (m *Manager) UpdateInstallProgress(ctx context.Context, h *models.Host, pro
 		return errors.Errorf("Can't set progress <%s> to host in status <%s>", progress.CurrentStage, swag.StringValue(h.Status))
 	}
 
-	if previousProgress.CurrentStage != "" && progress.CurrentStage != models.HostStageFailed {
-		// Verify the new stage is higher or equal to the current host stage according to its role stages array
-		stages := m.GetStagesByRole(h.Role, h.Bootstrap)
-		currentIndex := indexOfStage(progress.CurrentStage, stages)
+	var extra []interface{}
+	if progress.CurrentStage != models.HostStageFailed {
 
-		if currentIndex == -1 {
-			return errors.Errorf("Stages %s isn't available for host role %s bootstrap %s",
-				progress.CurrentStage, h.Role, strconv.FormatBool(h.Bootstrap))
-		}
-		if currentIndex < indexOfStage(previousProgress.CurrentStage, stages) &&
-			!m.allowStageOutOfOrder(h, progress.CurrentStage) {
-			return errors.Errorf("Can't assign lower stage \"%s\" after host has been in stage \"%s\"",
-				progress.CurrentStage, previousProgress.CurrentStage)
+		if previousProgress.CurrentStage != "" {
+			// Verify the new stage is higher or equal to the current host stage according to its role stages array
+			stages := m.GetStagesByRole(h.Role, h.Bootstrap)
+			currentIndex := m.IndexOfStage(progress.CurrentStage, stages)
+
+			if currentIndex == -1 {
+				return errors.Errorf("Stages %s isn't available for host role %s bootstrap %s",
+					progress.CurrentStage, h.Role, strconv.FormatBool(h.Bootstrap))
+			}
+			if currentIndex < m.IndexOfStage(previousProgress.CurrentStage, stages) &&
+				!m.allowStageOutOfOrder(h, progress.CurrentStage) {
+				return errors.Errorf("Can't assign lower stage \"%s\" after host has been in stage \"%s\"",
+					progress.CurrentStage, previousProgress.CurrentStage)
+			}
 		}
+
+		stages := m.GetStagesByRole(h.Role, h.Bootstrap)
+		currentIndex := m.IndexOfStage(progress.CurrentStage, stages)
```

Comment:
Is -1 a valid index stage because it can be returned from the function ?

Reply:
Same as https://github.com/openshift/assisted-service/pull/2208/files/9ab80aad5efd3881d8ec06c362c071a221118f90#r673712867